src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -591,7 +591,33 @@ private function getAdjacentRecordID($offset)
 
         $map = $list->limit($limit, $limitOffset)->column('ID');
         $index = array_search($this->record->ID, $map ?? []);
-        return isset($map[$index+$offset]) ? $map[$index+$offset] : false;
+        $position = $index+$offset;
+        $page = $currentPage;
+        $lastPosition = count($map) - 1;
```

Comment:
We don't need `$lasPosition` - we can use `$itemsPerPage` instead.
```suggestion
```

Reply:
I need a number of remaining items for the current page. `$itemsPerPage` has a constant value for each page.

Reply:
> `$lastPosition` on a page in the middle of all pages is ~`$itemsPerPage + 2`

Huh.... that sounds like a bug.... but not one this PR should aim to fix, to it's okay to leave this alone.

Reply:
Having looked into this a bit more, I can see that `$map` isn't just the items _for the current page_ - it's the items for the current page _plus_ the last item from the previous page _and_ the first item on the next page. I'm convinced we don't need this variable. See below.